doc/contrib/docs.rst

```diff
@@ -8,23 +8,73 @@ Documentation and Examples
   :backlinks: none
   :local:
 
-*********
-Documents
-*********
+*************
+Documentation
+*************
 * Python and C documentation is built using `Sphinx <http://www.sphinx-doc.org/en/master/>`_.
 * Each document is written in `reStructuredText <http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_.
-* You can build document locally to see the effect, by running
+* The documentation is the ``doc/`` directory.
+* You can build it locally using ``make html`` command.
 
-.. code-block:: bash
+   .. code-block:: bash
 
-  make html
+     make html
 
-inside the ``doc/`` directory.  The online document is hosted by `Read the Docs <https://readthedocs.org/>`__ where the imported project is managed by `Hyunsu Cho <https://github.com/hcho3>`__ and `Jiaming Yuan <https://github.com/trivialfis>`__.
+The online document is hosted by `Read the Docs <https://readthedocs.org/>`__ where the imported project is managed by `Hyunsu Cho <https://github.com/hcho3>`__ and `Jiaming Yuan <https://github.com/trivialfis>`__.
+
+===============================
+Build Docs on macOS using Conda
+===============================
+
+#. Create a conda environment.
+
+   .. code-block:: bash
+
+     conda create -n xgboost-docs --yes python=3.10
+
+   .. note:: Python 3.10 is required by `xgboost_ray <https://github.com/ray-project/xgboost_ray>`__ package.
+
+#. Activate the environment
+
+   .. code-block:: bash
+
+     conda activate xgboost-docs
+
+#. Install required packages (in the current environment) using ``pip`` command.
+   For some reason, it is currently not possible to install the required packages using ``conda``.
+
+   .. code-block:: bash
+
+     pip install -r requirements.txt
+
+#. (optional) Install graphviz
+
+   .. code-block:: bash
+
+     brew install graphviz
+
+#. Eventually, build the docs.
+
+   .. code-block:: bash
+
+     make html
+
+  You should see the following messages in the console:
+
+  .. code-block:: console
+
+    $ make html
+    sphinx-build -b html -d _build/doctrees   . _build/html
+    Running Sphinx v6.2.1
+    ...
+    The HTML pages are in _build/html.
+
+    Build finished. The HTML pages are in _build/html.
```

Comment:
A reference to the sphinx doc build would be great. There are many other options like output format, each with a different set of dependencies (like pdflatex for pdf).

Reply:
_A reference to the sphinx doc build would be great._ is already included in the `Documentation` section at the top of the page so I skip this. I added a note about the other options (`make help`).